### PR TITLE
chore: support the extended readiness check

### PR DESCRIPTION
**Pull request by kobim**
Following https://github.com/oauth2-proxy/oauth2-proxy/pull/1839, this will add the **non-breaking** support for the extended healthcheck as a `readinessProbe`.

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@kobim this is not enough, you need to bump also the chart version.

**Comment by pierluigilenoci** [comment]
@kobim if this PR is ready please mark it "Ready for review"

**Comment by pierluigilenoci** [comment]
@kobim could you please take a look?

**Comment by kobim** [comment]
@pierluigilenoci I'm still waiting to complete and merge https://github.com/oauth2-proxy/oauth2-proxy/pull/1839 before marking this as ready.

**Comment by pierluigilenoci** [comment]
@kobim the other PR is merged, can you please follow up on this?

**Review by pierluigilenoci — APPROVED** [review]

File: helm/oauth2-proxy/templates/deployment.yaml [hunk 1]
```diff
@@ -194,7 +194,7 @@ spec:
 {{- if .Values.readinessProbe.enabled }}
         readinessProbe:
           httpGet:
-            path: /ping
+            path: {{ gt .Chart.AppVersion "7.4.0" }}/ready{{ else }}/ping{{ end }}
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/templates/deployment.yaml:197]: The real software version is generated in this way:
```
{{ .Values.image.tag | default (printf "v%s" .Chart.AppVersion) }}
```

**Review by pierluigilenoci — CHANGES_REQUESTED** [review]

File: helm/oauth2-proxy/templates/_helpers.tpl [hunk 2]
```diff
@@ -101,3 +101,10 @@ Compute the redis url if not set explicitly.
 {{ fail "please set sessionStorage.redis.standalone.connectionUrl or enable the redis subchart via redis.enabled" }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Returns the version
+*/}}
+{{- define "oauth2-proxy.version" -}}
+{{ trimPrefix "v" (lower (.Values.image.tag | default (printf "v%s" .Chart.AppVersion))) }}
+{{- end -}}
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/templates/_helpers.tpl:110]: A new line here is missing.

**Review by pierluigilenoci — APPROVED** [review]

**Pull request merged**